---
title: "Tandem Repeats in Colorectal Cancer"
subtitle: "Characterization and Functional Analysis of Tandem Repeat"
author: "Matteo Delucchi"
date: "May 21, 2019"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{General Overview of Tandem Repeats}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---

```{r, echo = FALSE, message = FALSE}
knitr::opts_chunk$set(collapse = T, comment = "#>", fig.width=6)
options(tibble.print_min = 4L, tibble.print_max = 4L)
set.seed(123)
```
# Introduction
In Switzerland, colorectal cancer is among the top 3 leading cause for cancer-related mortality [1][krebsliga]. As the disease shows varying clinical outcomes and therapeutic responses, researchers try through personalized medicine to find individually optimal treatments.

The development of colorectal cancer (CRC) is accompanied by genetic and epigenetic mutations in onco- and tumor-suppressor genes which are supposed to cause CRC onset, progression and metastasis [2][Pancione2012] [3][Ewing2014].

Some clinical relevant biomarkers are today already used as prognostic and/or therapeutic predictive markers [4][Rosa2015].

[krebsliga]: https://www.krebsliga.ch/ueber-krebs/krebsarten/dickdarmkrebs/
[Pancione2012]: http://dx.doi.org/10.1155/2012/509348
[Ewing2014]: http://dx.doi.org/10.1136/flgastro-2013-100329
[Rosa2015]: https://doi.org/10.3892/or.2015.4108
[Kosinski2007]: https://doi.org/10.1073/pnas.0707210104

## Colorectal Cancer on a Molecular Level
Undifferentiated cells, such as colon stem progenitor cells, are located at the bottom of a crypt in healthy colons. They are able of self-renewal and pluripotency. 
During their live-cycle of about 14 days, they differentiate in all epithelial colon cell lineages (Paneth, goblet, enterocytes and enteroendocrine) while migrating from the crypt to the top of the villus leading into apoptosis [5][Kosinski2007]. 
The maintenance of genomic integrity is essential for proper cell differentiation in the colon. If it is lost, it's easy to accumulate multiple mutations through chromosomal instability [6][Pino2010], microsatellite instability [7][Boland2010], aberrant DNA methylation [8][Goel2007] and/or DNA repair defects leading to CRC.

Several altered molecular signaling pathways are involved in CRC onset, such as Wnt/APC/$\beta$-catenin, phosphoinositide 3-kinase (PI3K)/AKT/glycogen synthase kinase-3$\beta$ (GSK-3$\beta$), transforming growth factor-$\beta$ (TGF)-$\beta$/Smad, NF- Îºb or mismatch repair genes (MMR) [9][Rosa2015].


[Goel2007]: https://doi.org/10.1053/j.gastro.2006.09.018
[Pino2010]: https://doi.org/10.1053/j.gastro.2009.12.065
[Boland2010]: https://doi.org/10.1053/j.gastro.2009.12.064

### Wnt / $\beta$-catenin pathway
Upon secretion of Wnt ligand it binds to Frizzled (Fz) receptors. This results in the inactivation of the multifunctional kinase GSK-3$\beta$ and in the stabilisation of $\beta$-catenin. $\beta$-catenin is a E-cadherin cell-cell adhesion protein and transcriptional activator. If it is stabilized, it gets accumulated in the cytoplasm and is translocated into the nucleus. There it interacts with members of the lymphoid enhancer factor (LEF)/t-cell factor (TCF) and activates specific target genes.
In absence of Wnt-signal, $\beta$-catenin is degradated preventing its nuclear translocation. The ubiquination and phosphorylation of $\beta$-catenin is controlled through casein kinase 1 (CK1) and the APC/Axin/GSK-3$\beta$-complex.

Activation of the Wnt pathway through mutations in the APC gene is known to have a leading role in CRC pathogenesis. The protein product of APC is among the key components of $\beta$-catenin destruction leading to an accumulation of $\beta$-catenin which is then translocated to the nucleus where it mediates gene transcription through interaction with the LEF/TCF transcription factor family (LEF-1/LEF1, TCF-1/TCF7, TCF-3/TCF7L1, and TCF-4/TCF7L2) [10][Najdi2011].
The same outcome is expected, however, less frequently, through mutations in $\beta$-catenin and Axin2 [11][Morin1997] [12][Liu2000]

An increased concentration of $\beta$-catenin in the cytoplasm promotes cellular proliferation. 
Nuclear expression of $\beta$-catenin can be identified immunohistochemically but is not a unique factor for CRC. However, it's among one of the components of a diagnostic panel [13][Nikolouzakis2018].

[Najdi2011]: http://www.doi.org/10.4103/1477-3163.78111
[Morin1997]: http://www.doi.org/10.1126/science.275.5307.1787
[Liu2000]: https://doi.org/10.1038/79859
[Nikolouzakis2018]: https://doi.org/10.3892/or.2018.6330

#### Wnt-Singnaling and Colon Cancer
Wnt-signaling plays a key role in early cell fate decision in embryonic stem cells and is therefore under extensive investigation in cancer research. 
For example, the Wnt target gene *LGR5* is a multipotent stem cell marker in crypts. ASCL2, SOX9 and Paneth cell defensins are other examples of Wnt target genes controlling various steps in the differentiation process of stem cells.

In different cancer types the $\beta$-catenin dependent pathway is differentially regulated in normal tissue compared to cancerous tissue and often shows $\beta$-catenin stabilizing mutations [14][Polanski2000] [15][Rubinfeld1997] resulting in enhanced Wnt-signaling. Of many upstream and downstream components of $\beta$-catenin, APC and $\beta$-catenin are among the ones which show crucial roles in Wnt-signaling and cancer. 

Looking at mouse models with activated Wnt-pathway through *APC* mutations, revealed that mutations in APC alone are not sufficient for carcinogenic development. The Apc^min/+^ mouse models showed non-invasive intestinal tumors [16][Merritt1997]. For unfavorable carcinoma development an increased activation of proto-oncogenetic signaling pathways (i.e. *K-ras*) and additional mutations in *TGFBRII*, *SMAD4*, *TP53* may be necessary [17][Fearon1990].

Further, LEF/TCF transcription factors play a crucial role in CRC and may offer therapeutic opportunities [18][Najdi2011].
The recruitment of $\beta$-catenin to target genes in Wnt-signaling is depending on a $\beta$-catenin binding domain at the N-terminus of the LEF/TCF transcription factor. At the C-terminus, a sequence-specific DNA binding domain recognize Wnt response elements in target genes.
Interestingly, different LEF/TCF loci produce different protein isoforms through alternative splicing and different promotors which influence Wnt-signaling and can lead to cancer development [19][Cardigan2012] [20][Arce2006].
The expression pattern of LEF/TCF loci varies between normal and cancerous human colon tissue. In healthy colon cells, TCF-1 and TCF-4 are expressed and LEF-1 and TCF-3 are silent. It could be shown in mouse models, that a loss of TCF-4 depeletes stem cells and eliminates cell proliferation in the intestine [21][Korinek1998]. TCF-1 is suggested to be a feedback repressor of TCF-4 and cooperates with APC to suppress CRC.
In cancerous colon cells, TCF-1 and TCF-4 are still expressed (eventhough now as a full-length form) but also LEF-1 is produced which results in a maximal $\beta$-catenin interaction and hence in oncogenic Wnt-signaling.

[Polanski2000]: https://doi.org/10.1101/gad.14.15.1837
[Rubinfeld1997]: https://doi.org/10.1126/science.275.5307.1790
[Merritt1997]: https://dx.doi.org/10.1073%2Fpnas.94.25.13927
[Fearon1990]: https://doi.org/10.1016/0092-8674(90)90186-I
[Cardigan2012]: https://dx.doi.org/10.1101%2Fcshperspect.a007906
[Arce2006]: https://doi.org/10.1038/sj.onc.1210056
[Korinek1998]: https://doi.org/10.1038/1270
[Roose1999]: https://doi.org/10.1126/science.285.5435.1923

#### Short Tandem Repeats in Wnt-Pathway
We could show in [22][swissrepeat] that TRs in proteins cluster near the termini and are involved in protein-protein interaction as well as in transcription processes and DNA-binding. As shown in the section above, Wnt-signaling and the development of CRC highly depend on processes predestined for TR involvement. 
Therefore, we'll take a closer look at the LEF/TCF transcription factor. Specifically we'll investigate if we can detect TRs with the state of the art classification method [23][Schaper2012] in LEF/TCF transcription factors and describe their role in the Wnt/$\beta$-catenin pathway with respect to CRC.

But we not only investigate on LEF/TCF proteins, in fact we analyse proteins containing tandem repeats (TRs) which are expressed by CRC favorable and unfavorable genes [24][proteinatlasCRC] with a focus on the members of the Wnt/$\beta$-catenin pathway.

[swissrepeat]: TODO
[Schaper2012]: https://dx.doi.org/10.1093%2Fnar%2Fgks726
[proteinatlasCRC]: https://www.proteinatlas.org/humanproteome/pathology/colorectal+cancer#colorectal%20cancerunfavourable  


<!-- But we not only investigate on LEF/TCF proteins but also of *TGFBRII*, *SMAD4*, *TP53* [17][Fearon1990]. Furtheron, it is known, that different types of CRC are caused through a variety of different mutations and [Nojadeh et al.][Nojadeh2018] provides a collection of genes related to CRC: *APC*, *K-ras*, *DCC*, *P53*, *COX-2*, *BCL-2*, *MMR*-genes, *MUTYH*, *STK11/LKB1*, *MLH1*, *MSH2*, *MSH6*, *PMS2*.

[Nojadeh2018]: https://dx.doi.org/10.17179%2Fexcli2017-948
--> 




# Methods 

## Data Sources and Collection
All protein sequences were retrieved via the REST API from UniProt/Swiss-Prot Knowledgebase [23][uniprot] release 2019_04 for the reviewed entries of the Homo Sapiens proteome (UP000005640).

### Protein Sequence Origin
The names of CRC favorable and unfavorable genes were downloaded manually (Mai 2019) from the proteinatlas website [24][proteinatlasCRC] and saved as .tsv. 
The sequences of proteins which are expressed from CRC favorable and unfavorable genes, were retreived from Swiss-Prot by querying all proteins from each gene. Next, we checked the .fasta file for duplicated protein IDs, which requiered manual inspection. 

Analogously, the protein sequences for all proteins which are related to the Wnt-pathway in Swiss-Prot were retrieved and filtered for duplicates.

[uniprot]: https://doi.org/10.1093/nar/gky1049

## Tandem Repeat Detection \& Filtering
For the detection and filtering of TRs, we applied the open-source Python 3 Tandem Repeat Annotation Library (TRAL) [25][Schaper2015]. 
*De novo* detection of TRs in protein sequences was performed applying the detectors HHrepID, T-REKS, TRUST and XSTREAM followed by a refinement with profile Hidden Markov Models (HMM) implemented with HMMER [26][Eddy1998] to capture most of the TR units and their boundaries.
The TRs, were then filtered on a significance level of $\alpha = 0.05$ for a Likelyhood Ratio Test under the null hypothesis of random TR sequence evolution, a TR unit divergence of $d_{TR_{units}} < 0.1$ specifying the number of substitution site dating back to the most recent TR unit ancestor, 
a TR unit length of $l \leq 100$ specifying the number of noninsertion sites of the TR unit and 
a TR unit number of $n \geq 2.5$ specifying the number of noninsertion sites in the TR-MSA divided by $l$ [27][Schaper2012].

[Schaper2015]: https://doi.org/10.1093/bioinformatics/btv306
[Eddy1998]: https://doi.org/10.1093/bioinformatics/14.9.755

### Processing TRAL Results for Data Mining in R
The detected TR characteristics and their MSA are stored for each protein in a single .tsv and binary .pickle file. For further analysis, we join the results for all proteins in one file for each group (unfavorable, favorable, Wnt-pathway) containing only proteins which have TRs. 

The TR detection and annotation workflow is implemented in Python 3 and freely available on [Github](https://github.com/matteodelucchi/CRC_TRs.git). For reproduction of the reported results, change the code in order to use the provided datasets. To obtain the most recent datasets, just adapt `main.py` regarding your systems specification before sourcing.


## TR Characterization in CRC Associated Proteins 
To provide a reproducible framework of the TR characterization the following results including the data of the previous section is implemented as [R-package on github](https://github.com/matteodelucchi/TRAL-Result-Analysis.git) which can be installed like a regular R-package. Sourcing the vignette `GeneralOverviewofTandemRepeats.Rmd` allows an exact reproduction of the reported results. 

The evaluation was performed using
```{r sessioninfo, echo=FALSE}
sessionInfo()
```

```{r message=FALSE, warning=FALSE, include=FALSE, paged.print=FALSE}
base_path <- "/home/matteo/polybox/MSc_ACLS/master_thesis/TRALResultAnalysis/"
```

```{r Housekeeping, message=FALSE, warning=FALSE, collapse=TRUE}
library(TRALResultAnalysis)
tr_crcfavorable_path <- paste0(base_path, "inst/extdata/TRs_favorable_proteins_CRC_sp_l100.tsv")
tr_crcunfavorable_path <- paste0(base_path, "inst/extdata/TRs_unfavorable_proteins_CRC_sp_l100.tsv")
tr_wnt_path <- paste0(base_path, "inst/extdata/TRs_Wnt_proteins_CRC_sp_l100.tsv")
dest_file_sp <- paste0(base_path, "data/swissprot_human.tsv")
dest_file_kin <- paste0(base_path, "data/swissprot_human_kinome.tsv")
```

```{r load data sets, eval=TRUE, echo=FALSE, message=TRUE, warning=TRUE}
tr_fav <- load_tr_annotations(tr_crcfavorable_path)
tr_unfav <- load_tr_annotations(tr_crcunfavorable_path)
tr_wnt <- load_tr_annotations(tr_wnt_path)
```

To get deeper insights about the proteins containing TRs, more information is added from Swiss-Prot by a left join
```{r Download sp_all if it doesnt exist}
sp_url <- "https://www.uniprot.org/uniprot/?query=organism:%22Homo%20sapiens%20(Human)%20[9606]%22%20reviewed:yes&format=tab&columns=id,entry%20name,reviewed,protein%20names,genes,organism,length,virus%20hosts,encodedon,database(Pfam),interactor,comment(ABSORPTION),feature(ACTIVE%20SITE),comment(ACTIVITY%20REGULATION),feature(BINDING%20SITE),feature(CALCIUM%20BIND),comment(CATALYTIC%20ACTIVITY),comment(COFACTOR),feature(DNA%20BINDING),ec,comment(FUNCTION),comment(KINETICS),feature(METAL%20BINDING),feature(NP%20BIND),comment(PATHWAY),comment(PH%20DEPENDENCE),comment(REDOX%20POTENTIAL),rhea-id,feature(SITE),comment(TEMPERATURE%20DEPENDENCE)&sort=organism"

# Download the swissprot file only if it doesn't already exist.
# Uncomment this, if you want to use the most recent available data!
if(!file.exists(dest_file_sp)){
    download.file(sp_url, destfile = dest_file_sp)
}

sp_all_fav <- load_swissprot(dest_file_sp, tr_fav)
sp_all_unfav <- load_swissprot(dest_file_sp, tr_unfav)
sp_all_wnt <- load_swissprot(dest_file_sp, tr_wnt)

tr_unfav_sp <- merge(x = tr_unfav, y = sp_all_unfav, by = "ID", all.x = TRUE)
tr_fav_sp <- merge(x = tr_fav, y = sp_all_fav, by = "ID", all.x = TRUE)
tr_wnt_sp <- merge(x = tr_wnt, y = sp_all_wnt, by = "ID", all.x = TRUE)
```


# Results \& Discussion

In 243 genes associated with unfavorable CRC expressing 286 proteins we could *de novo* detect 617  TRs of which after filtering and clustering 106 TRs remained. In CRC favorable genes (352) 403 proteins are expressed, where we could *de novo* detect 486 TRs with 50 TRs beeing left after filtering and clustering.

Additionally, 644 proteins were associated in Swiss-Prot with the Wnt-pathway, where we could *de novo* detect 1353 TRs of which 199 TRs remained after filtering and clustering. 

## More TRs Were Detected in Proteins Associated With Unfavorable CRC Prognosis
```{r fraction of TR containing proteins}
# CRC favorable
length(unique(tr_fav_sp$ID))/403 # through unique() we ensure to count those proteins with >1 TR only once.

# CRC unfavorable
length(unique(tr_unfav_sp$ID))/286 

# Wnt pathway
length(unique(tr_wnt_sp$ID))
length(unique(tr_wnt_sp$ID))/644 
```
Of all proteins expressed by genes associated for beeing favorable or unfavorable for CRC, 10% and 23% contain at least one TR respectively. Analogously, we could detect TRs in 143 (22%) of the 644 proteins involved in the Human Wnt-pathway.

## Homo-TR Cluster Towards the N-Terminus
TRs generally cluster to the flanks of proteins with a tendency of domain-TRs to cluster near the N-terminus [TODO][swissrepeat].
```{r warning=FALSE}
TR_location(
  rbind(tr_fav_sp, tr_unfav_sp),
  byTRtype = TRUE)
```

In proteins expressed from CRC favorable and unfavorable genes, homo-TRs show a tendency for N-terminal location. 
Many of the homo-TR fall in the region allocated to signal sequences which tags the proteins for extracellular translocation through the cell membrane.
```{r warning=FALSE}
TR_location(tr_fav_sp, byTRtype = TRUE)
TR_location(tr_unfav_sp, byTRtype = TRUE)
```

Looking seperately at CRC favorable and unfavorable proteins it can be observed that favorable CRC proteins have a greater tendecy for N-terminal location of their homo-TRs. Which allows to hypothesize, that favorable CRC proteins may have a signal sequence and play extracellular roles more frequently than CRC unfavorable proteins.


It could be shown [TODO][Labaj2010] that N-terminal signal peptides are rich in Leucine residues. 
```{r}
# combine all TR from the three groups
tr_all <- rbind(tr_fav, tr_unfav, tr_wnt)

AAfreq_all <- AAfreq_in_TR(tr_all) 
AAfreq_all[base::order(AAfreq_all$aa_ratio, decreasing = TRUE),]

AAfreq_all_homo <- AAfreq_in_TR(tr_all[which(tr_all$l_type == "homo"),]) 
AAfreq_all_homo[base::order(AAfreq_all_homo$aa_ratio, decreasing = TRUE),]

AAfreq_all[which(AAfreq_all$aa == "L"),]
AAfreq_all_homo[which(AAfreq_all_homo$aa == "L"),]
```
We combine all datasets and count the amino acid frequency and compare it to the amino acid frequency in homo-TRs which doesn't show much difference for Lysine. 
However, we can see, that Alanine is most frequent (14%) followed by Serine (12%) and Proline (12%) in CRC and Wnt-/$\beta$-catenin pathway associated proteins.

```{r TODO is this necessary}
# (AAfreq_fav <- AAfreq_in_TR(tr_fav))
# (AAfreq_unfav <- AAfreq_in_TR(tr_unfav))
# cor.test(AAfreq_fav$aa_ratio, AAfreq_unfav$aa_ratio, method = "spearman")
```


```{r}
AAratio_vs_Disorderpropensity(tr_all, plot_title = "CRC & Wnt-Pathway Associated Proteins")
```

Plotting the amino acid frequency ratio of TRs in CRC and Wnt-/$\beta$-catenin pathway associated proteins against the amino acid disorderpropensity we can see that more amino acids with high disorder propensity are part of TR in CRC and Wnt-/$\beta$-catenin pathway associated proteins - with exception of Lysine.
```{r}
AAratio_vs_Disorderpropensity(sp_overall = TRUE)
```

This trend is consistent with the overall amino acid frequency of all known proteins - amino acids with higher disorderpropensity appear in TRs significantly more frequent.

[Labaj2010]: https://doi.org/10.1111/j.1742-4658.2010.07720.x


## Only Few Wnt-pathway Proteins Are Associated With CRC.
Proteins with TRs from the Wnt-pathway seem to be distinct from the proteins retrieved by genes. 
```{r fav in wnt-pathway}
# CRC favorable proteins in Wnt pathway
tr_fav_sp$protein_name[which(tr_fav_sp$ID %in% tr_wnt_sp$ID)]
```
3 Proteins which are associated with favorable CRC prognosis are invovled in the Wnt-pathway and
```{r unfav in wnt-pathway}
# CRC unfavorable proteins in Wnt pathway
tr_unfav_sp$protein_name[which(tr_unfav_sp$ID %in% tr_wnt_sp$ID)]
```
5 with unfavorable prognosis.

## Nucleic Acid Binding Proteins Show Many Repeat Regions
The number of detected TRs exceeds the number of analysed proteins which suggests that there are proteins with more than one TR-region.
```{r many TRs}
# CRC favorable proteins
table(table(tr_fav$ID))

# CRC unfavorable proteins
table(table(tr_unfav_sp$ID))

# Wnt-pathway proteins
table(table(tr_wnt_sp$ID))
```
Nevertheless, most of the proteins, contain only one TR-region we detect up to 11 different TR-regions i.e. in some CRC unfavorable associated proteins.

In proteins associated with unfavorable CRC prognosis we found in general more proteins with $\geq 4$ TRs compared to the group of unfavorable or Wnt-pathway proteins.
```{r multiple TR regions in unfavorable prots I}
protein_id_by_number_of_TR(tr_unfav_sp, 11)
eleven_TRs <- tr_unfav_sp[which(tr_unfav_sp$ID == protein_id_by_number_of_TR(tr_unfav_sp, 11)[[1]]),]
unique(eleven_TRs$protein_name)
unique(eleven_TRs$prot_function)
```
For example the splicing factor protein SFR19_HUMAN (ID: Q9H7N4) shows 11 TR-regions. 

```{r multiple TR regions in unfavorable prots II}
protein_id_by_number_of_TR(tr_unfav_sp, 5)
five_TRs <- tr_unfav_sp[which(tr_unfav_sp$ID == protein_id_by_number_of_TR(tr_unfav_sp, 5)[1]),]
unique(five_TRs$protein_name)
unique(five_TRs$prot_function)
five_TRs <- tr_unfav_sp[which(tr_unfav_sp$ID == protein_id_by_number_of_TR(tr_unfav_sp, 5)[2]),]
unique(five_TRs$protein_name)
unique(five_TRs$prot_function)
```
And Forkhead-related transcription factor 3 (ID: Q12948) and CLK4 splicing factor appear to have 5 TR-regions where the former is promoting cell growth inhibition and is involved in cell migration [TODO refrence to section below where it's more investigated].

## Pathogenic Tandem Repeat Types Prevail
```{r}
summary(tr_fav_sp$l_effective)
unique(tr_fav_sp$l_type)
```

```{r}
summary(tr_unfav_sp$l_effective)
unique(tr_unfav_sp$l_type)
```

```{r}
summary(tr_wnt_sp$l_effective)
unique(tr_wnt_sp$l_type)
```
In all three groups of proteins we could detect homo- micro- and small TRs of which homo- and micro-TRs are well known to be disease causing. We could further find domain-TR in CRC favorable proteins and in proteins associated to the Wnt-pathway.

```{r}
summary(tr_fav_sp$n_effective)
summary(tr_unfav_sp$n_effective)
summary(tr_wnt_sp$n_effective)
```

```{r}
summary(tr_fav_sp$total_repeat_length)
summary(tr_unfav_sp$total_repeat_length)
summary(tr_wnt_sp$total_repeat_length)
```
All TR-regions have a mean number of repeat units of 7 units per TR-region. The mean TR-region length of CRC associated proteins is about 9 amino acids in a range of 6 up to 70, 53 for CRC favorable and unfavorable proteins respectively.
In Wnt-pathway related proteins we could detect a mean repeat unit number of 7 and a TR-region length of 17 ranging from 6 to 864 amino acids.
```{r}
tr_wnt_sp[which(tr_wnt_sp$total_repeat_length == max(tr_wnt_sp$total_repeat_length)),]
```
The clinical breast cancer marker Mucin-1 is the protein with the longest TR-region. It's expression is increased in most carcinoma cells i.a. colorectal cancer [TODO][Nakamori1994] [TODO][Zeng2015]. 
It's single TR-region falls in the extracellular domain and provides many sites for O- and N-linked glycosylation. The amount of gylcosylation is increased in breast cancer cells [TODO][Mueller1999] which is caused through decreased homophilic adhesion [TODO][Makiguchi1996], increased invasiveness [TODO][Suwa1998] and protection of cytotoxic T-cell reactions. Through its TR-region MUC1 can bind to ICAM-1. It could be shown, that the large TR size contributes to increased binding properties [TODO][Hayashi2001] and is responsible in cell migration and metastases.
The cytoplasmic tail stabilizes $\beta$-catenin through direct interaction [TODO][Wen2003] which can result in different cancer phenotypes [TODO][Lillehoj2007].
[TODO reference to results below]

[McGuckin1995]: https://doi.org/10.1016/0046-8177(95)90146-9
[Guddo1998]: https://www.ncbi.nlm.nih.gov/pubmed/9677444
[Utsunomiya1998]: http://clincancerres.aacrjournals.org/content/4/11/2605
[Nakamori1994]: https://doi.org/10.1016/0016-5085(94)90592-4
[Zeng2015]: https://dx.doi.org/10.1371%2Fjournal.pone.0138049
[Mueller1999]: http://dx.doi.org/10.1074/jbc.274.26.18165
[Makiguchi1996]: https://dx.doi.org/10.1111%2Fj.1349-7006.1996.tb00252.x
[Suwa1998]: https://doi.org/10.1002/(SICI)1097-0215(19980504)76:3%3C377::AID-IJC15%3E3.0.CO;2-8
[Hayashi2001]: https://doi.org/10.1159/000051917
[Wen2003]: https://doi.org/10.1074/jbc.M304333200
[Lillehoj2007]: https://doi.org/10.1016/j.bbamcr.2007.04.009

```{r}
tr_wnt_sp$protein_name[which(tr_wnt_sp$total_repeat_length > 15)]
unique(tr_wnt_sp$prot_function[which(tr_wnt_sp$total_repeat_length > 15)])
```
Wnt-pathway associated proteins with long TR-regions are generally involved in transcriptional regulation. But also others, such as the WD repeat-containing protein 26 which negatively regulated the Wnt-pathway contains long repeat regions.

```{r}
tr_fav_sp$protein_name[which(tr_fav_sp$total_repeat_length > 15)]
unique(tr_fav_sp$prot_function[which(tr_fav_sp$total_repeat_length > 15)])
```
Similarly proteins associated with favorable prognosis of CRC containing long TR-regions show beside transcriptional regulatory functions also roles in regulation of alternative splicing.

```{r}
tr_unfav_sp$protein_name[which(tr_unfav_sp$total_repeat_length > 15)]
unique(tr_unfav_sp$prot_function[which(tr_unfav_sp$total_repeat_length > 15)])
```
Interestingly, the Numb-like protein (ID: Q9Y6R0, Numbl) contains a long (20 AA) poly-Q homo-TR. Those kind of TR are prone site for diseases caused by tandem repeat disorders (TRD). Numbl is further a negative regulator of NF-$\kappa$-B pathway. Negative regulation of NF-$\kappa$-B pathway is known as a source of CRC through interaction with Wnt/$\beta$-catenin pathway. One possible interaction leading to CRC is through NF-$\kappa$-B upregulated Lef1-expression which enhances transcriptional activity of TCF/LEF [REF][MA2016].

[Ma2016]: https://dx.doi.org/10.3389%2Ffimmu.2016.00378


## Binding of LEF/TCF to $\beta$-Catenin is Mediated Through Tandem Repeats
```{r}
# Show only a selection of variables
sel_var <- c("ID", "begin", "msa_original", "repeat_region_length", "l_type", "prot_name", "protein_name", "gene_names", "prot_function")
tr_wnt_sp[which(tr_wnt_sp$ID == "Q9UJU2"),sel_var]
```
In LEF1 we found a significant poly-G TR-region at the N-terminal end which falls in the binding region with $\beta$-catenin1 (positions 1-66 based on similarity [TODO][LEF1-Uniprot]). *LEF1* has been identified as beeing only expressed in CRC cells but not in healthy colon tissue. However, it could be detected that different splicing variants in alteri result in the loss of the $\beta$-catenin binding site which acts as a transcriptional repressor often expressed in healthy colon epithelial cells [TODO][Hovanes2001].

```{r}
tr_wnt_sp[which(tr_wnt_sp$ID == "Q9HCS4"),sel_var]
tr_wnt_sp[which(tr_wnt_sp$ID == "P36402"),sel_var]
```
A similar motif could be detected in TCF-3 and TCF-7 which have also a poly-G TR-region near the N-terminus offering a presumable binding region for $\beta$-catenin1 [TODO][Barker1999]. Through this binding region, TCF-3 can mediate $\beta$-catenin degradation [TODO][Lee2001]. 

[LEF1-Uniprot]: https://www.uniprot.org/uniprot/Q9UJU2#family_and_domains
[Hovanes2001]: https://www.nature.com/articles/ng0501-53
[Barker1999]: https://doi.org/10.1016/S0065-230X(08)60783-6  
<!-- https://doi.org/10.1002/ijc.24802 --> 
[Lee2001]: https://dx.doi.org/10.1083%2Fjcb.200102074

## $\beta$-Catenin Destruction Complex Not Only Relies On Tandem Repeats
The $\beta$-catenin destruction complex comprises different proteins, such as adenomatosis polyposis coli (APC), Axin, protein phosphatase (PP2A), glycogen synthase kinase 3 (GSK3) and casein kinase 1$\alpha$ (CK1$\alpha$).

### APC $\beta$-Catenin Binding Domain Contains Tandem Repeats
```{r}
tr_wnt_sp[grepl("APC", tr_wnt_sp$prot_name),sel_var]
```
The polyS homo-TR region in the APC protein sequence was the only repeat region which could be detected, eventhough there are structurally verified armadillo repeats near the N-terminal end of the protein sequence. 
However, in the very similar APC-like protein, we could detect a micro-TR in a region associated with $\beta$-catenin binding necessary for its degradation [TODO][Nakagawa1998].


[Nakagawa1998]: http://cancerres.aacrjournals.org/content/58/22/5176

### GSK
```{r}
tr_wnt_sp[grepl("GSK", tr_wnt_sp$prot_name),sel_var]
```
In GSK3A and GSK3B we could detect a polyG homo-TR at N-terminal proximity and a micro TR near the C-terminus respectively. 

### Casein Kinase I
```{r}
tr_wnt_sp[grepl("KC", tr_wnt_sp$prot_name),sel_var]
tr_fav_sp[grepl("KC", tr_fav_sp$prot_name),sel_var]
tr_unfav_sp[grepl("KC", tr_unfav_sp$prot_name),sel_var]
```
With the applied filtering approach, no TRs were found in Casein kinase I isoform epsilon of Wnt-pathway associated or CRC favorable proteins. However, in the set of proteins CRC unfavorable, we found a small TR in the C-terminal cytoplasmic region of ATP-sensitive inward rectifier potassium channel 8 protein.

### Axin Offers A Prone Site For Tandem Repeat Disorder
```{r}
tr_wnt_sp[grepl("AXIN", tr_wnt_sp$prot_name),sel_var]
```
Mutations in AXIN1 and AXIN2 are involved with CRC pathogenesis [TODO][Liu2000]. In Axin-2 overlapping with the compositionally biased histidine region, a polyH homo-TR could be detected which falls in the region interacting with armadillo repeats in $\beta$-catenin.

TRD in this region may destabilize the binding of $\beta$-catenin to the protein-complex with Axin-2 resulting in a accumulation of $\beta$-catenin followed by a transcriptional activation of the Wnt/$\beta$-catenin target genes which is known to be a leading cause CRC.


## Dishevelled Proteins
```{r}
tr_wnt_sp[grepl("DVL", tr_wnt_sp$prot_name),sel_var]
```
Dishevelled proteins bind to the C-terminus of frizzeled proteins promoting their internalisation upon Wnt-signaling [TODO][Zang2009].
DVL-2 shows a polyP homo-TR near it's C-terminal end which could to our knowledge not be brought in contribution with Frizzled-binding.

FOXK1 and FOXK2 are positively transcriptionally regulating the Wnt-pathway and interact with DVL proteins. FOXK1/2 are reported to be highly expressed in CRC and correlate with nuclear localisation of DVL proteins [TODO][Wang2015].
```{r}
tr_wnt_sp[grepl("FOX", tr_wnt_sp$prot_name),sel_var]
tr_fav_sp[grepl("FOX", tr_fav_sp$prot_name),sel_var]
tr_unfav_sp[grepl("FOX", tr_unfav_sp$prot_name),sel_var]
```
In each protein FOXK1 and FOXK2 we could detect two TR-regions: A polyA homo-TR near the N-terminus and a micro-TR near the C-terminus. In FOXK1 the polyA homo-TR falls in the interaction domain with SIN3A. FOXK1 toghether with SIN3A cooperatively regulate cell cycle progression [TODO][Yang2002]. 

Forkhead box protein C1 (FOXC1) in contrast is part of the CRC unfavorable associated proteins. FOXC1 acts as a transcriptional activator where the crucial domains are located at the C- and N-termini. Most of the detected TRs fall exactely in these regions responsible for transcriptional activation [TODO][Berry2002]. The C-terminal end 366-553 is the proposed binding region of a degradation factor [TODO][Berry2006] and the region 47-553 could be shown to bind to PITX2 [TODO][Berry2006.2] supporting the idea of TRs acting as keyplayers in protein-protein interactions. To our knowledge, no structural interaction data is available to further prove this hypothesis.

Interestingly, in both FOXK and FOXC proteins the forkhead box didn't show any TRs - they are rather clustered around the forkhead box.


```{r}
tr_wnt_sp[which(tr_wnt_sp$ID == "O00358"),sel_var]
tr_unfav_sp[which(tr_unfav_sp$ID == "P10070"),sel_var]
```
*FOXE1* is one of the prognostic colorectal cancer genes and can indirectly activate the Wnt/$\beta$-catenin pathway [TODO][Ma2019] and shows a central poly-A TR-region.

[Wang2015]: https://doi.org/10.1016/j.devcel.2015.01.031
[Yang2002]: http://dx.doi.org/10.1016/S0092-8674(02)00810-3 
[Ma2019]: https://doi.org/10.1002/jcp.28399
[Berry2002]:  https://doi.org/10.1074/jbc.M110266200
[Berry2006]: https://doi.org/10.1074/jbc.M513629200
[Berry2006.2]: https://doi.org/10.1093/hmg/ddl008

## Wnt Receptor Complex
The Wnt Receptor Complex is made up of Wnt, Frizzled (Fz), G-protein coupled receptors (GPCRs), lipoprotein receptor-related protein (LRP), receptor tyrosine kinase (RTK) and ROR.

### Frizzled
```{r}
tr_wnt_sp[grepl("FZ", tr_wnt_sp$prot_name),sel_var]
```
Frizzled membran proteins are receptors of Wnt activating Dishevelled proteins.
In FZD5 and FZD8 we could detect polyL and polyG homo-TR near the C- and N-terminus respectively. 
The polyL TR in FZD5 is near but not within the Frizzled domain and on the opposite terminus from the DVL binding site.
In FZD8, the polyG TR-region is extracellular forming a helical structure which is in topological proximity to the Wnt binding site.

### WNT
```{r}
tr_wnt_sp[grepl("WNT", tr_wnt_sp$prot_name),sel_var]
```
In WNT2B and WNT6 polyL homo-TRs are located near the N-terminus in a signal peptide domain which is the important part of interaction with signal recognition particles in the process of nuclear translocation.

### LRP
```{r}
tr_wnt_sp[grepl("LRP", tr_wnt_sp$prot_name),sel_var]
```
LRP5 has a polyL homo-TR in its signal peptide at the N-terminus and a polyS homo-TR near the C-terminus. The later is also found at a similar location in LRP6. To our knowledge there is not much known today about those regions at the C-terminus.
In LRP1 the N-terminal polyL homo-TR is located in a helical transmembran peptide in a possible Low-density lipoprotein receptor region. In the canonical Wnt-pathway only LRP5 and LRP6 are reported to be involved.

[Zang2009]: http://dx.doi.org/10.1038/nchembio.152

### Receptor Tyrosine Kinase \& Tyrosine-protein kinase transmembrane receptor
```{r}
tr_wnt_sp[grepl("PTK", tr_wnt_sp$prot_name),sel_var]
tr_wnt_sp[grepl("ROR", tr_wnt_sp$prot_name),sel_var]
```
The inactive tyrosine-protein kinase 7 (PTK7) and Tyrosine-protein kinase transmembrane receptor (ROR2) are mainly involved in the $\beta$-catenin independent planar cell polarity Wnt-pathway. PTK7 plays an important role in cancer cell invasion [TODO][Golubkov2010].
ROR2 is shown to suppress the Wnt/$\beta$-catenin signal and is therefore also an interesting player in the fight against cancer [TODO][Bainbridge2014].
We couldn't detect in both of those Kinase any TRs as we couldn't in CKI and GSK3 neither. It is thus possible to hypothesise that protein kinases tend to have few TR in general.

[Golubkov2010]: http://dx.doi.org/10.1074/jbc.M110.165159
[Bainbridge2014]: https://dx.doi.org/10.1371/journal.pone.0102695

## Kinases Show No Tandem Repeat Depletion
As we couldn't detect any TRs in certain protein kinases of the Wnt/$\beta$-catenin pathway, we're interested if protein kinases generally lack TRs.
We therefore filter the Human Proteome for protein kinases (enzyme commission numbers 2.7.10, 2.7.11, 2.7.12, 2.7.13, 2.7.14, 2.7.99).
```{r}
# url to UniProt/Swiss-Prot querying enzyme commissions for protein kinases
url_kin <- "https://www.uniprot.org/uniprot/?query=ec:2.7.10.-%20OR%20ec:2.7.11.-%20OR%20ec:2.7.12.-%20OR%20ec:2.7.13.-%20OR%20ec:2.7.14.-%20OR%20ec:2.7.99.-&format=fasta&sort=score&fil=proteome:UP000005640%20AND%20reviewed:yes%20AND%20organism:%22Homo%20sapiens%20(Human)%20[9606]%22"
sp_kinIDs <- load_kinome(url = url_kin, path = dest_file_kin, OnlyIDs = TRUE)

# No. of protein kinases in Human Proteome
length(sp_kinIDs)
length(sp_kinIDs) / nrow(sp_all_fav)
# No. protein kinases in Wnt-Pathway
sum(sp_kinIDs %in% tr_wnt$ID)
sum(sp_kinIDs %in% tr_wnt$ID) / nrow(tr_wnt)
# No. protein kinases in CRC favorable proteins
sum(sp_kinIDs %in% tr_fav$ID)
# No. protein kinases in CRC unfavorable proteins
sum(sp_kinIDs %in% tr_unfav$ID)

(sum(sp_kinIDs %in% tr_fav$ID) + sum(sp_kinIDs %in% tr_unfav$ID))/(nrow(tr_fav)+nrow(tr_unfav))
```
Of the 483 protein kinases from the Human proteome 11 have TRs and are associated with the Wnt-pathway. 7 protein kinases have TRs and appear in the set of CRC related proteins.
To put this in relations, of the whole human proteome 2% are protein kinases. From proteins with TRs, 4% belong to the enzyme class of protein kinases in CRC associated proteins and 6% in Wnt-pathway proteins. 
```{r}
dest_file_kinext <- "/home/matteo/polybox/MSc_ACLS/master_thesis/TRALResultAnalysis/data/swissprot_human_extkinome.tsv"
url_extkin <- "https://www.uniprot.org/uniprot/?query=ec:2.7.-.-%20AND%20reviewed:yes%20AND%20organism:%22Homo%20sapiens%20(Human)%20[9606]%22%20AND%20proteome:up000005640&format=fasta&sort=score"
sp_extkinIDS <- load_kinome(url = url_extkin, path = dest_file_kinext, OnlyIDs = TRUE)
# No. of protein kinases in Human Proteome
length(sp_extkinIDS)
length(sp_extkinIDS) / nrow(sp_all_fav)
# No. protein kinases in Wnt-Pathway
sum(sp_extkinIDS %in% tr_wnt$ID)
sum(sp_extkinIDS %in% tr_wnt$ID) / nrow(tr_wnt)
# No. protein kinases in CRC favorable proteins
sum(sp_extkinIDS %in% tr_fav$ID)
# No. protein kinases in CRC unfavorable proteins
sum(sp_extkinIDS %in% tr_unfav$ID)

(sum(sp_extkinIDS %in% tr_fav$ID) + sum(sp_extkinIDS %in% tr_unfav$ID))/(nrow(tr_fav)+nrow(tr_unfav))
```
These findings don't support our hypothesis and even considering all phosphorous transfering enzymes (EC 2.7.-.-) doesn't show a depletion of TRs.


# Conclusion
- Often TR-region in lox complexity regions / yet not characterized regions of the protein (see Forkhead. Where the TR-regions encompass the main functional region; OR Kinase. Do they never have TRs?)

- Extracellular or membranlocated proteins show a tendency for homo-TRs in their signal peptide sequence (see Wnt and LRP).

- TRs (domain) act in PPI (MUC1)

- The interaction of proteins is increased and mediated through TRs

- critically reflect the results, as the sequences are often processed such that different forms of the proteins are available.
























# NOTES \& THOUGHTS
## Filter the top 20 CRC genes:
```{r}
top20_unfav_genes <- c("LRCH4", "POFUT2", "CLK3", "EGFL7", "DPP7", "HSH2D", "ASB6", "SPAG4", "EXOC3L4", "HSPA1A", "PAQR6", "FAM69B", "CRACR2B", "ARHGAP4", "NPDC1", "DAPK1", "CNPY3", "ARL8A", "INAFM1", "RHBDD2")
(getTRbyGene(tr_sp = tr_unfav_sp, genes = top20_unfav_genes))

top20_fav_genes <- c("RBM3", "NOL11", "USP53", "TEX2", "HOOK1", "ZYG11B", "HSPA8", "DLAP", "SORT1", "DDX46", "FBXO7", "ABCD3", "NGLY1", "PARS2", "CLCC1", "AP3B1", "PRPSAP1", "PSMA5", "GRSF1", "CD274")
(getTRbyGene(tr_fav_sp, genes = top20_fav_genes))
```


## compare to results from paulina
She found those genes resp. proteins to have TRs
```{r}
lina_top6_genes <- c("CDX2", "CLK3", "CNPY3", "CRACR2B", "DPP7", "HSH2D")
lina_top6_prots <- c("Q9NQX5", "H3BVF8", "A0A0C4DFY8", "Q8N4Y2", "R4GMV4", "Q96JZ2")
```

Not all of them appear in the list of genes from \url{https://www.proteinatlas.org/humanproteome/pathology/colorectal+cancer#colorectal%20cancerunfavourable} (22. Mai 2019)
```{r}
lina_top6_genes %in% top20_unfav_genes
```

And we couldn't find TRs in all proteins mentioned in her thesis. Only in this subset:
```{r}
(getTRbyGene(tr_sp = tr_unfav_sp, genes = lina_top6_genes)[1])
```

# Comparative analysis of favorable TRs and unfavorable
TODO!!!!!!!!!

# Disorderpropensity Meta Data
## Disorderpropensity per Protein
TODO: Add what we want here

Start with determining the Amino Acid distribution of the TR Sequence
```{r}
# (aa_freq_TR <- AAfreq_in_TR(Q9UGU0))
```

We sort them according their disorderpropensity. You find more information in Uversky et al. ( http://www.tandfonline.com/doi/full/10.4161/idp.24684).
```{r}
# # Not mentioned in Uversky's paper: "B", "O", "U", "Z", "X". These guys might need to fit in with the rest (if possible, as some of them represent multiple aa.)
# aa_order_promoting_to_disorder_promoting = c("C", "W", "I", "Y", "F", "L", "H", "V", "N", "M", "R", "T", "D", "G", "A", "K", "Q", "S", "E", "P", "B", "O", "U", "Z", "X")
# # Sort AA according to their disorder promoting potential
# aa_freq_TR <- aa_freq_TR[match(aa_order_promoting_to_disorder_promoting, aa_freq_TR$aa),]
# colnames(aa_freq_TR) <- c("aa_freq_tr", "aa", "aa_ratio_tr")
```


We then load the dataset with information about the amino acid frequency of all swissprot proteins and the disorderpropensity of each amino acid. This data is included in the package and is generated through the script "/data-raw/create_AAfreqSP.R". Rerunning this script, updates the amino acid frequency of swissprot proteins.
```{r}
# data("AAfreqSP")
```


We can now download the whole protein sequence and compare the Amino acid frequency compared to that of the overall protein
```{r}
# prot_seq <- download_prot_sequence("Q9UGU0")
# aa_freq_prot <- AAfreq_in_prot(prot_seq)
# (aa_freq_prot <- aa_freq_prot[match(aa_order_promoting_to_disorder_promoting, aa_freq_prot$aa),])
# colnames(aa_freq_prot) <- c("aa_freq_sp", "aa", "aa_ratio_sp")
```

Plot our amino acid abundancy in the TRs of this protein against the amino acid abundancy in the whole protein, we see that AA with medium to high disorderpropensity fall in the TR sequence.
```{r}
# # we combine the datasets
# aa_freq_prot <- aa_freq_prot[1:20,]
# disorderpropensity <- AAfreqSP$disorderpropensity
# df <- cbind(aa_freq_TR[1:20,], disorderpropensity, aa_freq_prot)
# df <- df[ , !(names(df) %in% c("aa"))]
# 
# p <- ggplot(df, aes(x= aa_ratio_sp, y = aa_ratio_tr, size = disorderpropensity))+
#   geom_point()+
#   labs(x= "AA Background Frequency",
#        y = "AA Frequency in TRs")+
#   guides(size=guide_legend(title="Disorderpropensity"))+
#   theme_minimal()
# p <- beautifier(p, x.axis.text.angle = 0)
# p
```


## Disorderprotpensity per Set of Proteins



